lib/hawkular/inventory/entities_v4.rb

```diff
@@ -82,6 +82,15 @@ def initialize(hash)
       @_hash = hash.dup
     end
 
+    def children_by_type(type, recursive = false)
+      selected_children = []
+      @children.each do |child|
+        selected_children << child if child.type.id == type
+        selected_children.push(*child.children_by_type(type, recursive)) if recursive
```

Comment:
No way to do it iteractively with a `flat_map`? Something like this might work:

```ruby
selected_children = @children.select { |x| x.type.id == type }

if recursive
  selected_children.flat_map! { |x| x.children_by_type(type, recursive) }
end

selected_children
```